src/tools/rbd/action/Group.cc

```diff
@@ -721,38 +727,127 @@ int execute_group_snap_list(const po::variables_map &vm,
   if (f) {
     f->open_array_section("group_snaps");
   } else {
+    t.define_column("ID", TextTable::LEFT, TextTable::LEFT);
     t.define_column("NAME", TextTable::LEFT, TextTable::LEFT);
     t.define_column("STATUS", TextTable::LEFT, TextTable::RIGHT);
   }
 
-  for (auto i : snaps) {
-    std::string snap_name = i.name;
-    int state = i.state;
-    std::string state_string;
-    if (RBD_GROUP_SNAP_STATE_INCOMPLETE == state) {
-      state_string = "incomplete";
+  for (const auto& snap : snaps) {
+    auto state_string = get_group_snap_state_name(snap.state);
+    if (f) {
+      f->open_object_section("group_snap");
+      f->dump_string("id", snap.id);
+      f->dump_string("snapshot", snap.name);
+      f->dump_string("state", state_string);
+      f->close_section();
     } else {
-      state_string = "ok";
+      t << snap.id << snap.name << state_string << TextTable::endrow;
     }
+  }
+
+  if (f) {
+    f->close_section();
+    f->flush(std::cout);
+  } else if (snaps.size()) {
+    std::cout << t;
+  }
+  return 0;
+}
+
+int execute_group_snap_info(const po::variables_map &vm,
+                            const std::vector<std::string> &ceph_global_args) {
+  size_t arg_index = 0;
+  std::string pool_name;
+  std::string namespace_name;
+  std::string group_name;
+  std::string group_snap_name;
+
+  int r = utils::get_pool_generic_snapshot_names(
+    vm, at::ARGUMENT_MODIFIER_NONE, &arg_index, at::POOL_NAME, &pool_name,
+    &namespace_name, GROUP_NAME, "group", &group_name, &group_snap_name, true,
+    utils::SNAPSHOT_PRESENCE_REQUIRED, utils::SPEC_VALIDATION_FULL);
+  if (r < 0) {
+    return r;
+  }
+
+  at::Format::Formatter formatter;
+  r = utils::get_formatter(vm, &formatter);
+  if (r < 0) {
+    return r;
+  }
+  Formatter *f = formatter.get();
+
+  librados::Rados rados;
+  librados::IoCtx io_ctx;
+  r = utils::init(pool_name, namespace_name, &rados, &io_ctx);
+  if (r < 0) {
+    return r;
+  }
+
+  librbd::RBD rbd;
+  librbd::group_snap_info2_t group_snap;
+  r = rbd.group_snap_get_info(io_ctx, group_name.c_str(),
+                              group_snap_name.c_str(), &group_snap);
+  if (r < 0) {
+    std::cerr << "rbd: failed to show group snapshot: "
+              << cpp_strerror(r) << std::endl;
+    return r;
+  }
+
+  std::vector<std::string> pool_names(group_snap.image_snaps.size());
+  for (size_t i = 0; i < group_snap.image_snaps.size(); i++) {
+    int64_t pool_id = group_snap.image_snaps[i].pool_id;
+    r = rados.pool_reverse_lookup(pool_id, &pool_names[i]);
     if (r < 0) {
+      std::cerr << "rbd: error looking up pool name for pool_id=" << pool_id
+                << ": " << cpp_strerror(r) << std::endl;
       return r;
```

Comment:
`rbd group image ls` produces a `"<missing image pool $POOL_ID>"` string instead of failing in this case and for a good reason: if a user deletes a pool with one of the group's images, we still want to be able to get information about that group or its snapshots.  We should do the same here.

It also occurs to me that it would be better to make `pool_names` a map from pool ID to pool name rather than a vector of pool names.  That way we wouldn't be storing duplicate strings (a group is likely to have all of its images come from a single pool) and it would be immune to sorting.

Reply:
> it would be better to make `pool_names` a map from pool ID to pool name rather than a vector of pool names

... or we could do exactly what `execute_list_images()` does and perform the lookup inside the loop, eliminating a separate data structure entirely.

Reply:
Now does lookup inside the loop. Uses a map to not redo look up of pool name.

Reply:
See https://github.com/ceph/ceph/pull/57797#discussion_r1681716952.

Reply:
I will create a tracker ticket for the issues with the `group` set of CLI commands when the pool of a group's image is removed.

Reply:
https://tracker.ceph.com/issues/67021

Reply:
Thanks for looking into this!